pkg/rbac/userData.go

```diff
@@ -153,9 +153,32 @@ func (user *UserDataCache) userHasAllAccess(ctx context.Context, cache *Cache) (
 
 		cache.shared.mcCache.lock.Lock()
 		defer cache.shared.mcCache.lock.Unlock()
-		user.ManagedClusters = cache.shared.managedClusters
+		user.ManagedClusters = map[string]struct{}{"*": {}}
 		user.clustersCache.updatedAt = time.Now()
 		user.csrCache.err, user.nsrCache.err, user.clustersCache.err = nil, nil, nil
+		klog.V(5).Infof("User %s with uid %s has access to all resources.",
+			user.userInfo.Username, user.userInfo.UID)
+
+		return true, nil
+	} else if user.userAuthorizedListSSAR(ctx, impersClientSet,
+		"get", "searches", "searches/allManagedData") {
+		user.csrCache.lock.Lock()
+		defer user.csrCache.lock.Unlock()
+		user.CsResources = []Resource{}
```

Comment:
**Only for discussion, no changes needed from this comment**

When the user is authorized to `searches/allManagedData` it means that they won't get any local resources even if authorized. 
This won't help the ACM Admin scenario as I hoped.  If we want to cover this scenario in the future, we could do it with a new subresource like `searches/inherit-local-bypass-managed`, which would implement this new behavior.

**The current implementation optimizes the Global Search user scenario which is all we need right now.**